pint/quantity.py

```diff
@@ -79,6 +78,262 @@ def wrapped(self, *args, **kwargs):
+            stripped_args, stripped_kwargs = convert_to_consistent_units(first_input_units,
+                                                                         *args, *kwargs)
+        else:
+            # Match all input args/kwargs to input_units, or if input_units is None, simply
+            # strip units 
+            stripped_args, stripped_kwargs = convert_to_consistent_units(input_units, *args,
+                                                                         *kwargs)
+
+        # Determine result through base numpy function on stripped arguments
+        result_magnitude = func(*stripped_args, **stripped_kwargs)
+
+        if output_unit is None:
+            # Short circuit and return magntidue alone
+            return result_magnitude
+        elif output_unit == "match_input":
+            result_unit = first_input_units
+        elif output_unit == "sum":
+            result_unit = (1 * first_input_units + 1 * first_input_units).units
+        elif output_unit == "prod":
+            product = 1
+            for x in args_and_kwargs:
+                product *= x
+            result_unit = product.units
+        elif output_unit == "delta":
+            result_unit = (1 * first_input_units - 1 * first_input_units).units
+        elif output_unit == "delta,div":
+            product=(1 * first_input_units - 1 * first_input_units).units
+            for x in args_and_kwargs[1:]:
+                product /= x
+            result_unit = product.units
+        elif output_unit == "variance":
+            result_unit = ((1 * first_input_units + 1 * first_input_units)**2).units
+        else:
+            raise ValueError('Output unit method {} not understood'.format(output_unit))
+
+        return first_input_units._REGISTRY.Quantity(result_magnitude, result_unit)
+
+@implements(np.meshgrid)
+def _meshgrid(*xi, **kwargs):
+    # Simply need to map input units to onto list of outputs
+    input_units = (x.units for x in xi)
+    res = np.meshgrid(*(x.m for x in xi), **kwargs)
+    return [out * unit for out, unit in zip(res, input_units)]
+
+@implements(np.full_like)
+def _full_like(a, fill_value, dtype=None, order='K', subok=True, shape=None):
+    # Make full_like by multiplying with array from ones_like in a
+    # non-multiplicative-unit-safe way
+    if isinstance(fill_value, Quantity):
+        return fill_value._REGISTRY.Quantity(
+            np.ones_like(a, dtype=dtype, order=order, subok=subok, shape=shape) * fill_value.m,
+            fill_value.units)
+    else:
+        return (np.ones_like(a, dtype=dtype, order=order, subok=subok, shape=shape)
+                * fill_value)
+
+@implements(np.interp)
+def _interp(x, xp, fp, left=None, right=None, period=None):
+    # Need to handle x and y units separately
+    (x, xp, period), _ = unwrap_and_wrap_consistent_units(x, xp, period)
+    (fp, right, left), output_wrap = convert_to_consistent_units(fp, left, right)
```

Comment:
this looks like you swapped `unwrap_and_wrap_consistent_units` and `convert_to_consistent_units` (and maybe add `None` to the arguments of `convert_to_consistent_units`?)